airbyte-cdk/python/airbyte_cdk/sources/declarative/requesters/error_handlers/http_response_filter.py

```diff
@@ -68,7 +71,11 @@ def matches(self, response_or_exception: Optional[Union[requests.Response, Excep
             if isinstance(response_or_exception, requests.Response):
                 error_message = self._create_error_message(response_or_exception)
             error_message = error_message or default_error_message
-            failure_type = default_mapped_error_resolution.failure_type if default_mapped_error_resolution else FailureType.system_error
+
+            if self.failure_type:
+                failure_type = self.failure_type
+            else:
```

Comment:
nit: I think this block would be easier to read as
```
if self.failure_type:
   ...
elif default_mapped_error_resolution:
   ...
else:
   ...
```

Reply:
updated